mods/taskbar-icon-numberer.wh.cpp

```diff
@@ -0,0 +1,468 @@
+            if (hex.length() == 9) { // #AARRGGBB
+                color.A = static_cast<uint8_t>(std::wcstoul(hex.substr(1, 2).c_str(), nullptr, 16));
+                color.R = static_cast<uint8_t>(std::wcstoul(hex.substr(3, 2).c_str(), nullptr, 16));
+                color.G = static_cast<uint8_t>(std::wcstoul(hex.substr(5, 2).c_str(), nullptr, 16));
+                color.B = static_cast<uint8_t>(std::wcstoul(hex.substr(7, 2).c_str(), nullptr, 16));
+            } else { // #RRGGBB
+                color.A = 255;
+                color.R = static_cast<uint8_t>(std::wcstoul(hex.substr(1, 2).c_str(), nullptr, 16));
+                color.G = static_cast<uint8_t>(std::wcstoul(hex.substr(3, 2).c_str(), nullptr, 16));
+                color.B = static_cast<uint8_t>(std::wcstoul(hex.substr(5, 2).c_str(), nullptr, 16));
+            }
+        } catch (...) {
+            color.A = 255; color.R = 255; color.G = 255; color.B = 255;
+        }
+    } else {
+        color.A = 255; color.R = 255; color.G = 255; color.B = 255;
+    }
+    return color;
+}
+
+enum class NumberPosition {
+    topLeft,
+    topRight,
+    bottomLeft,
+    bottomRight
+};
+
+struct {
+    NumberPosition numberPosition;
+    int numberSize;
+    std::wstring numberColor;
+    std::wstring backgroundColor;
+    bool showOnlyRunning;
+    int maxNumbers;
+} g_settings;
+
+std::atomic<bool> g_taskbarViewDllLoaded;
+std::atomic<bool> g_unloading;
+std::mutex g_overlayMutex;
+std::unordered_set<void*> g_numberedButtons;
+
+// Hook function pointers
+using TaskListButton_get_IsRunning_t = HRESULT(WINAPI*)(void* pThis, bool* running);
+TaskListButton_get_IsRunning_t TaskListButton_get_IsRunning_Original;
+
+using TaskListButton_UpdateVisualStates_t = void(WINAPI*)(void* pThis);
+TaskListButton_UpdateVisualStates_t TaskListButton_UpdateVisualStates_Original;
+
+// Utility functions
+FrameworkElement FindChildByName(FrameworkElement element, PCWSTR name) {
+    if (!element) return nullptr;
+    
+    try {
+        int childrenCount = Media::VisualTreeHelper::GetChildrenCount(element);
+        for (int i = 0; i < childrenCount; i++) {
+            auto child = Media::VisualTreeHelper::GetChild(element, i).try_as<FrameworkElement>();
+            if (child && child.Name() == name) {
+                return child;
+            }
+        }
+    } catch (...) {
+        // Ignore errors
+    }
+    return nullptr;
+}
+
+bool TaskListButton_IsRunning(FrameworkElement taskListButtonElement) {
+    if (!TaskListButton_get_IsRunning_Original || !taskListButtonElement) {
+        return false;
+    }
+    
+    try {
+        bool isRunning = false;
+        TaskListButton_get_IsRunning_Original(
+            winrt::get_abi(taskListButtonElement.as<winrt::Windows::Foundation::IUnknown>()),
+            &isRunning);
+        return isRunning;
+    } catch (...) {
+        return false;
+    }
+}
+
+void CreateNumberOverlay(FrameworkElement taskListButtonElement, int number) {
```

Comment:
I noticed that this function is called for each taskbar item on each update operation (e.g. mouse hover), which seems inefficient. What do you think about adding an optimization such that the overlay is updated only when necessary? You can get the current text and create a new element only if there's no label yet or if the number changed.

Even if the number changes, if the label already exists, you don't need to recreate it each time.